.golangci.yml

```diff
@@ -29,6 +29,14 @@ linters:
   disable-all: true
 
 linters-settings:
+  govet:
+    enable:
+      - nilness
+      - unusedwrite
+    # enable-all: true
+    # disable:
+    #   - fieldalignment
```

Comment:
fieldalignment is also somewhat intereseting. Enabling it would mean 100+ updates to current struct. How many of these cases actually have practical benefits is not fully clear to me yet. This could be considered but needs more analysis before.

Shadow could catch some issues but currently there are quite a lot of places where we intentionally use same variable name and changing it could hurt readability.

Reply:
With `fieldalignment`, wouldn't that mean we couldn't keep our structs ordered for readability? Unless the memory usage is incredibly dramatic, I'd prefer to allow ordering/grouping of struct fields to be done to keep readability of the code.